azurerm_sentinel_alert_rule_scheduled: support incident configuration (#8564)
terraform-providers/terraform-provider-azurerm: website/docs/r/sentinel_alert_rule_scheduled.html.markdown

* `trigger_threshold` - (Optional) The baseline number of query results generated, combined with `trigger_operator`, setting alert threshold of this Sentinel Scheduled Alert Rule.

---

> A `grouping_configuration` block supports the following:

Reviewer: shouldn't this be under the incident configuration block desription?

Author: The order here follows alphabetic order (as implemented in the scaffolding tool by Tom). The referencing verb is properly handled - if you look at the statement where the `grouping` is referenced in `incident_configuration` block, it uses statement:

> `grouping` - (Optional) A `grouping` block as defined above.

Reviewer: As much as that may be the case, wouldn't it make more sense to have the sub blocks documented after the block they are contained in? otherwise the user will scroll down and up continually and there is no structure to the documentation?

Author: Agreed, I will update it.

> * `enabled` - (Required) Whether this grouping configuration enabled for the incidents created from alerts triggered by this Sentinel Scheduled Alert Rule?

Reviewer: ```suggestion
* `enabled` - (Required) Enable grouping incidents created from alerts triggered by this Sentinel Scheduled Alert Rule. Defaults to [true|false]
```

> * `lookback_duration` - (Optional) ISO8601 lookback duration within which the group is to be created. Defaults to `PT5M`.

Reviewer: This description doesn't make sense?

* `reopen_closed_incident` - (Optional) Whether to re-open closed matching incidents? Defaults to `false`.

> * `entity_matching_method` - (Optional) Grouping matching method. Possible values are `All`, `Custom` and `None`. Defaults to `None`.

Reviewer: ```suggestion
* `entity_matching_method` - (Optional) The method used to group incidents. Possible values are `All`, `Custom` and `None`. Defaults to `None`.
```

> * `group_by_entities` - (Optional) A list of entity types to group by (**only when `entity_matching_method` is `Custom`**). Possible values are `Account`, `Host`, `Url`, `Ip`.

Reviewer: ```suggestion
* `group_by_entities` - (Optional) A list of entity types to group by only when the `entity_matching_method` is `Custom`. Possible values are `Account`, `Host`, `Url`, `Ip`.
```
